examples/runCustomTask/runCustomTask.ts

```diff
@@ -21,21 +62,86 @@ The question is: "${USER_INPUT}"
     id: '1',
     totalCost: 0,
     stopped: false,
-    currentStageIndex: 0,
     onChanged: async () => {
       console.log('Changed ' + task.progress);
     },
     executionStage: '',
     progress: 0,
+    stageTargetProgress: 0,
     startTime: Date.now(),
     logContent: '',
     rejectTask: () => {},
     resolveTask: () => {},
   });
 
   console.log('Running task ...');
-  await mutateRunTask(task);
-  console.log(task.logContent);
+  await mutateRunTaskStages(task, async (task) => {
+    mutateStartStage(task, 'Stage 1', 0.5 / 3);
+    mutateAppendToLog(task, 'Stage 1' + '\n');
+    mutateEndStage(task);
+
+    mutateStartStage(task, 'Stage 2', 0.5 / 3);
+    mutateAppendToLog(task, 'Stage 2' + '\n');
+    mutateEndStage(task);
+
+    mutateStartStage(task, 'Stage 3: Choose Strategy', 0.5 / 3);
+    mutateAppendToLog(task, 'Stage 3: Choose Strategy' + '\n');
```

Comment:
as we use this `0.5 / 3` value a few times assign it to const please and the reuse it there